### PR TITLE
cmd: support defaulting to the user's preferred payment method

### PR DESCRIPTION
**Pull request by pete-woods**

### CODE REVIEW

File: cmd/snap/cmd_buy.go [hunk 1]
```diff
@@ -124,26 +124,43 @@ func buySnap(opts *store.BuyOptions) error {
...
+		if !(stringResponse == "" && index > 0) {
+			// If the user answered and also has no preferred method, we need to parse the response.
+			index, err = strconv.Atoi(stringResponse)
+			if err != nil {
+				return fmt.Errorf(i18n.G("cannot buy snap %q: invalid payment method selection %q"), snap.Name, stringResponse)
+			}
 
-		if index <= 0 || index > len(paymentInfo.Methods) {
-			return fmt.Errorf(i18n.G("cannot buy snap %q: unknown payment method selection %d"), snap.Name, index)
+			if index <= 0 || index > len(paymentInfo.Methods) {
```
Comment (zyga) [inline cmd/snap/cmd_buy.go]: I know that index is `i + 1` (so 1 based) so this test is okay but it may be confusing. Does index have to be 1 based?

**Comment by zyga** [comment]
Looks okay, just one question/suggestion on the index variable

File: cmd/snap/cmd_buy.go (hunk 1 above)
Comment (pete-woods) [inline cmd/snap/cmd_buy.go]: Well it's 1 based, as that makes more sense to the user (well to me at least).

File: cmd/snap/cmd_buy.go [hunk 2]
```diff
@@ -110,26 +110,43 @@ func buySnap(opts *store.BuyOptions) error {
...
+
 		fmt.Fprintln(w, i18n.G("\tSelection\tDescription"))
 		for i, method := range paymentInfo.Methods {
-			fmt.Fprintf(w, "\t%d\t%s\n", i+1, method.Description)
+			preferred := ""
+			if method.Preferred {
+				preferred = "*"
+				index = i + 1
+			}
+			fmt.Fprintf(w, "%s\t%d\t%s\n", preferred, i+1, method.Description)
```
Comment (chipaca) [inline cmd/snap/cmd_buy.go]: here I would've defaulted `preferred` to a space, and used `%s%d``, so the output is more compact.

**Comment by chipaca** [comment]
:+1: One very, very, minor, ignorable nit.

And a question: what does this do if stdout is a pipe?

**Pull request merged**